cmd/kubeadm/app/cmd/cmd.go

```diff
@@ -80,8 +83,14 @@ func NewKubeadmCommand(in io.Reader, out, err io.Writer) *cobra.Command {
 		},
 	}
 
-	cmds.ResetFlags()
+	namedFlagSets := &cliflag.NamedFlagSets{}
```

Comment:
Sanity question, do we need to touch kubeadm at all. I don't see any place where they import `component-base/logs`.

I think we are going pretty deep into kubeadm, making harder to review. I would suggest thinking about if we can split this PR into smaller and ensure that the most important changes are prioritized. It might take multiple stages to get all changes in, but without this it will require root level approver which there are not many.

Reply:
> Sanity question, do we need to touch kubeadm at all. I don't see any place where they import component-base/logs.

In ` globalflag. AddGlobalFlags ` function  import `component-base/logs`
https://github.com/kubernetes/kubernetes/blob/2342ac3cfa46e4e69005fec6461012fdf9fc6dc3/staging/src/k8s.io/component-base/cli/globalflag/globalflags.go#L34

> I think we are going pretty deep into kubeadm, making harder to review. I would suggest thinking about if we can split this PR into smaller and ensure that the most important changes are prioritized. It might take multiple stages to get all changes in, but without this it will require root level approver which there are not many.

Yes, you are right, I will change it, try to change as little code as possible, make PR smaller.